File: services/catalog-service/src/main/java/com/tap2eat/catalog/dtos/response/internal/ValidateOrderResponse.java
package com.tap2eat.catalog.dtos.response.internal;

import java.math.BigDecimal;
import java.util.List;

public record ValidateOrderResponse(
        String restaurantId,
        String branchId,
        boolean valid,
        List<ValidatedOrderItemResponse> items,
        BigDecimal subtotal,
        BigDecimal total
) {
}

File: services/order-service/Integrations/Catalog/Dtos/ValidateOrderResponse.cs
namespace OrderService.Integrations.Catalog.Dtos;

public sealed class ValidateOrderResponse
{
    public string RestaurantId { get; set; } = string.Empty;

    public string BranchId { get; set; } = string.Empty;

    public bool Valid { get; set; }

    public List<ValidatedOrderItemResponse> Items { get; set; } = [];

    public decimal Subtotal { get; set; }

    public decimal Total { get; set; }
}
